homeassistant/components/ruuvitag_ble/sensor.py

```diff
@@ -0,0 +1,158 @@
+"""Support for RuuviTag sensors."""
+from __future__ import annotations
+
+from typing import Optional, Union
+
+from sensor_state_data import (
+    DeviceKey,
+    SensorDeviceClass,
+    SensorDeviceInfo,
+    SensorUpdate,
+    Units,
+)
+
+from homeassistant import config_entries, const
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataProcessor,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+    PassiveBluetoothProcessorCoordinator,
+    PassiveBluetoothProcessorEntity,
+)
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+SENSOR_DESCRIPTIONS = {
+    (SensorDeviceClass.TEMPERATURE, Units.TEMP_CELSIUS): SensorEntityDescription(
+        key=f"{SensorDeviceClass.TEMPERATURE}_{Units.TEMP_CELSIUS}",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=const.TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    (SensorDeviceClass.HUMIDITY, Units.PERCENTAGE): SensorEntityDescription(
+        key=f"{SensorDeviceClass.HUMIDITY}_{Units.PERCENTAGE}",
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=const.PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    (SensorDeviceClass.PRESSURE, Units.PRESSURE_HPA): SensorEntityDescription(
+        key=f"{SensorDeviceClass.PRESSURE}_{Units.PRESSURE_HPA}",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=const.PRESSURE_HPA,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    (
+        SensorDeviceClass.VOLTAGE,
+        Units.ELECTRIC_POTENTIAL_MILLIVOLT,
+    ): SensorEntityDescription(
+        key=f"{SensorDeviceClass.VOLTAGE}_{Units.ELECTRIC_POTENTIAL_MILLIVOLT}",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=const.ELECTRIC_POTENTIAL_MILLIVOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    (
+        SensorDeviceClass.SIGNAL_STRENGTH,
+        Units.SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    ): SensorEntityDescription(
+        key=f"{SensorDeviceClass.SIGNAL_STRENGTH}_{Units.SIGNAL_STRENGTH_DECIBELS_MILLIWATT}",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=const.SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    (SensorDeviceClass.COUNT, None): SensorEntityDescription(
+        key=f"{SensorDeviceClass.COUNT}_movement",
+        device_class=SensorDeviceClass.COUNT,
+        native_unit_of_measurement=None,
```

Comment:
```suggestion
```

Reply:
Well, the line is unnecessary, sure, since the default set in the dataclass is `None`, but it doesn't hurt. I'd rather keep it there so it's Explicit Is Better Than Implicit (via the Zen of Python), and symmetric with the other registry entries.

Reply:
@bdraco The "wasn't visible" bug wasn't/isn't caused by this though, but rather https://github.com/home-assistant/core/blob/ebd86abf59733f8570c5c4bc1a3d7d23f88aa59e/homeassistant/components/ruuvitag_ble/sensor.py#L116
which was copied from `tilt-ble` (since all of its measurements do have dimensions). That's fixed in a future revision, I just wanted to know whether `None` is correct for a dimensionless quantity.

Reply:
![Screenshot 2022-11-08 at 16 29 55](https://user-images.githubusercontent.com/58669/200591362-45143431-4fc7-4b1a-b17f-8e53bb87473b.png)
`None` seems to be correct alright.

Reply:
```suggestion
```

`None` is correct, but our standard is to not set it if its `None`

Reply:
Changed.